Use OpenMP imported targets instead of injecting compiler flags globally.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -390,10 +390,8 @@
 option(WITH_OPENMP "Compile with parallelization support using OpenMP" OFF)
 if(WITH_OPENMP)
   find_package(OpenMP REQUIRED)
-  if(OPENMP_FOUND)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS} -DWITH_OPENMP")
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${OpenMP_C_FLAGS} -DWITH_OPENMP")
-  endif()
+  add_compile_definitions(WITH_OPENMP)
+  link_libraries(OpenMP::OpenMP_C OpenMP::OpenMP_CXX)
 endif()
 
 # enabling Threads support if requested
